components/p3a/brave_p3a_service.cc

```diff
@@ -339,8 +213,8 @@ BraveP3ALogStore::LogForJsonMigration BraveP3AService::Serialize(
 bool
 BraveP3AService::IsActualMetric(base::StringPiece histogram_name) const {
```

Comment:
basically can we move this function to p3a_metrics.h as well? 
since recently it is possible to create a constexpr flat_set, so we can do it directly, instead of having an array + NoDestructor thing

Reply:
That would be nice! Do you mean `base::fixed_flat_set`? That says it requires a sorted initialiser, and it looks like `std::sort` isn't constexpr until C++20, so I think we'd have to give up the interior sections and put the P2A metrics in the middle. Worthwhile?

Reply:
The docs were unclear; `base::MakeFixedFlatSet` appears to [sort the initializer](https://source.chromium.org/chromium/chromium/src/+/refs/tags/101.0.4951.34:base/containers/fixed_flat_set.h;l=93). I'll try that.

Reply:
perhaps we could put the the list itself into .cc and expose only `IsP3AMetric` or something from the .h file.

Reply:
I tried that first, but in addition to `IsP3AMetric` we need an enumerator for `BraveP3AService::InitCallbacks`. Implementing that on a wrapper class seemed like a lot of code for single-use encapsulation, and passing the BraveP3AService object in is an even worse layer violation.

Looking at it again, maybe passing a `base::RepeatingCallback` to a RegisterMetricObservers() wouldn't be too bad.